### DISCUSSION TITLE
How to synchronize property in nested component?

### DISCUSSION THREAD

**Discussion by dbwodlf3** [lit/lit#2663] [Q&A] [answered]
```ts
//....
class ComponentA {
    //....
    @property()
    dataB="";

   //....
   render(
        return html`
            <componentA>
                <componentB dataB=${this.dataB}> <componentB>
            </componentA>
    `
    )
}

// ....
class ComponentB {
    // ...
    @property()
    dataB="";
}
```
It is effected to change dataB in ComponentA, but it is not to changed dataB in ComponentB.
How to synchronize dataB in ComponentA and dataB in ComponentB?

**Comment by p-ob** [comment, marked as answer]
Lit's documentation does call out "properties down, events up" (https://lit.dev/docs/composition/component-composition/#passing-data-up-and-down-the-tree). So emitting an event from the child component when the data has been modified, so the parent can react to that is the common approach.

**Reply by dbwodlf3** [reply]
Oh. really thank you...
